DesktopEdge/Models/ZitiService.cs

```diff
@@ -121,13 +121,33 @@ public class ZitiService {
 
 		public Visibility TimerVisibility {
 			get {
-				return (TimeoutRemaining>-1&&TimeoutRemaining<1260) ? Visibility.Visible : Visibility.Collapsed;
+				return (TimeoutCalculated > -1&& TimeoutCalculated < 1260&& TimeoutCalculated > 0) ? Visibility.Visible : Visibility.Collapsed;
+			}
+			set { }
+		}
+
+		public int TimeoutCalculated { 
+			get {
+				if (this.TimeoutRemaining == -1|| TimeoutRemaining == 0) return this.TimeoutRemaining;
+				else {
+					TimeSpan t = (DateTime.Now - this.TimeUpdated);
+					int timeout = this.Timeout - (int)Math.Floor(t.TotalSeconds);
+					if (timeout < 0) timeout = 0;
+					return timeout;
+				}
+			}
+			set { }
+		}
+
+		public Visibility MfaVisibility {
+			get {
+				return (TimeoutCalculated > -1 && TimeoutCalculated == 0) ? Visibility.Visible : Visibility.Collapsed;
 			}
 			set { }
 		}
 		public int TimerWidth {
 			get {
-				return (TimeoutRemaining > -1 && TimeoutRemaining < 1260) ? 20 : 0;
+				return (TimeoutCalculated > -1 && TimeoutCalculated < 1260) ? 20 : 0;

```

Comment:
same comment - i think this logic : 
```
TimeoutCalculated > -1 && TimeoutCalculated < 1260
```
should be encapsulated in a function... this if fails if TimeoutCalculated is > 1260 as well as if it's 0

Reply:
I encapsulate in a property like this, I prefer that for UI Models. It has been updated to 1200 per the other conversation